### PR TITLE
Lexc improvements

### PR DESCRIPTION
**Pull request by elmurod1202**

### CODE REVIEW

File: apertium-uzb.uzb.lexc [hunk 1]
```diff
@@ -40745,3 +40800,5557 @@ Amerika% Qoʻshma% Shtatlari:Amerika% Qoʻshma% Shtatlari  NP-TOP ; ! "" ! El++
...
+Abramovich:Abramovich  NP-COG-MF ; ! "" ! El++
+Abrasevich:Abrasevich  NP-COG-MF ; ! "" ! El++
+Absalamov:Absalamov  NP-COG-MF ; ! "" ! El++
+Absalamova:Absalamova  NP-COG-MF ; ! "" ! El++
+Abselamov:Abselamov  NP-COG-MF ; ! "" ! El++
+Abselamova:Abselamova  NP-COG-MF ; ! "" ! El++
+Acimambetov:Acimambetov  NP-COG-MF ; ! "" ! El++
+Acimambetova:Acimambetova  NP-COG-MF ; ! "" ! El++
+Acimovich:Acimovich  NP-COG-MF ; ! "" ! El++
+Adagamov:Adagamov  NP-COG-MF ; ! "" ! El++
```
Comment (jonorthwash) [inline apertium-uzb.uzb.lexc]: There's a special lexicon for names ending in -ov and -ova in other Turkic transducers (check e.g. kaz and tat).  Could you copy that way of doing this?  Same for -vich, -ovna, etc.

File: apertium-uzb.uzb.lexc [hunk 2]
```diff
@@ -40745,3 +40800,5557 @@ Amerika% Qoʻshma% Shtatlari:Amerika% Qoʻshma% Shtatlari  NP-TOP ; ! "" ! El++
...
+SSC:SSC  NP-AL ; ! "" ! El++
+Super% G4:Super% G4  NP-AL ; ! "" ! El++
+Svoboda:Svoboda  NP-AL ; ! "" ! El++
+Tarjimon:Tarjimon  NP-AL ; ! "" ! El++
+Tribuna:Tribuna  NP-AL ; ! "" ! El++
+TSK:TSK  NP-AL ; ! "" ! El++
+Twitter:Twitter  NP-AL ; ! "" ! El++
+Ukrenergo:Ukrenergo  NP-AL ; ! "" ! El++
+Ukrinform:Ukrinform  NP-AL ; ! "" ! El++
+Ukrtelekom:Ukrtelekom  NP-AL ; ! "" ! El++
```
Comment (jonorthwash) [inline apertium-uzb.uzb.lexc]: Many of these can be `<np><org>`—better than NP-AL.

File: apertium-uzb.uzb.lexc [hunk 3]
```diff
@@ -3392,12 +3397,12 @@ bariton:bariton N1 ; !
 barkamol:barkamol A2 ; ! 
 barkastalik:barkastalik N1 ;  ! 
 barkash:barkash N1 ; ! 
-barqaror:barqaror A1 ; ! 
-barqarorlik:barqarorlik N1 ;  ! 
+barqaror:barqaror A1 ; ! "stabil"
```
Comment (jonorthwash) [inline apertium-uzb.uzb.lexc]: "stable"

File: apertium-uzb.uzb.lexc [hunk 4]
```diff
@@ -35708,7 +35740,7 @@ elfurush:elfurush N1 ; ! 0.9999999987189747
 elchilik:elchilik N1 ; ! 0.9999999995851189
 elchilik:elchilik N1 ; ! 0.9999999999999409
 elchixona:elchixona N1 ; ! "embassy"
-elchi:elchi N1 ; ! "postman"
+elchi:elchi N1 ; ! "ambassador"
 elchi:Elchi N1 ; ! "ambassador"
 Elchi:Elchi N1 ; ! "ambassador"
```
Comment (jonorthwash) [inline apertium-uzb.uzb.lexc]: Why do we need three versions of this?  Apertium should handle capitalisation differences automatically.

File: apertium-uzb.uzb.lexc [hunk 5]
```diff
@@ -35900,6 +35932,7 @@ erkatoy:erkatoy N1 ; ! 0.9999965658440407
 erka:erka N1 ; ! 0.9995437488829819
 erkinlik:erkinlik N1 ; ! 0.999999999995661
 erkin:erkin A1 ; ! "free"
+erkin:erkin ADV ; ! "freely"
```
Comment (jonorthwash) [inline apertium-uzb.uzb.lexc]: This isn't covered by `<adj><advl>` readings?

**Review by jonorthwash — CHANGES_REQUESTED** [review]

File: apertium-uzb.uzb.lexc (hunk 1 above)
Comment (elmurod1202) [reply apertium-uzb.uzb.lexc]: Good, I was thinking of having a separate lexicons for that, gonna copy from kaz(or tat) then. For now, I just wanted to have the same tag as in tur.lexc, later regroup again.  I also am planning to import ANT-M and ANT-Fs from Uzbek names book.

File: apertium-uzb.uzb.lexc (hunk 2 above)
Comment (elmurod1202) [reply apertium-uzb.uzb.lexc]: Sure, will sort them out again.

File: apertium-uzb.uzb.lexc (hunk 3 above)
Comment (elmurod1202) [reply apertium-uzb.uzb.lexc]: Hah, looks like even-though I was born after the Uzbekistan's independence, deep down inside, I have some Russian instincts :D . Fixed it.

File: apertium-uzb.uzb.lexc (hunk 4 above)
Comment (elmurod1202) [reply apertium-uzb.uzb.lexc]: Does it handle well?. Ex: names of the months in Uzbek were already there in capital letters, but I saw all month names in lowercases not-analysed when calculating uzb hfst coverage. So I was second-guessing about it.

File: apertium-uzb.uzb.lexc (hunk 5 above)
Comment (elmurod1202) [reply apertium-uzb.uzb.lexc]: Apparently, it is. So it is not really necessary to add them separately?. I mean, yes when analysing uzbek text, it is handled, but is it also handled properly when there is an entry at bidix: `<l>erken("adv")</l>  <r>erkin("adv")</r>`?

File: apertium-uzb.uzb.lexc (hunk 4 above)
Comment (jonorthwash) [reply apertium-uzb.uzb.lexc]: It only handles it well if you put the default form in lower case. Uppercase forms should only be added to the dictionary when that's the only correct way of writing a word (like names).

File: apertium-uzb.uzb.lexc (hunk 5 above)
Comment (jonorthwash) [reply apertium-uzb.uzb.lexc]: For translation, use `erkin<adj><advl>` too.

**Comment by jonorthwash** [comment]
The new names seem to have extra spaces.  Did you check that it compiles before committing?

**Comment by elmurod1202** [comment]
> The new names seem to have extra spaces. Did you check that it compiles before committing?

Oh, my bad, just fixed it. I was quick to push changes as soon as it compiled without an error, not noticing the spaces.

File: apertium-uzb.uzb.lexc (hunk 5 above)
Comment (elmurod1202) [reply apertium-uzb.uzb.lexc]: I guess I got a grasp of this now (looking at examples from kaz-tat), fixed it.

File: apertium-uzb.uzb.lexc (hunk 4 above)
Comment (elmurod1202) [reply apertium-uzb.uzb.lexc]: Noted. I fixed this and entries alike this one

File: apertium-uzb.uzb.lexc (hunk 2 above)
Comment (elmurod1202) [reply apertium-uzb.uzb.lexc]: Sorted them out, they They became NP-ALs, NP-ORGs and an NP-TOP both in tur.lexc, tur-uzb.dix and respective translations in uzb.lexc as well.

**Comment by jonorthwash** [comment]
I'm curious about these forms; what are they?
```
{+ad:ad NP-AL ; !+}
{+al:al NP-AL ; !+}
{+am:am NP-AL ; !+}
{+bei:bei NP-AL ; !+
{+ibn:ibn NP-AL ; !+}
{+la:la NP-AL ; !+}
{+le:le NP-AL ; !+}
{+les:les NP-AL ; !+}
{+lès:lès NP-AL ; !+}
```

**Comment by jonorthwash** [comment]
> I'm curious about these forms; what are they?
> 
> ```
> {+ad:ad NP-AL ; !+}
> {+al:al NP-AL ; !+}
> {+am:am NP-AL ; !+}
> {+bei:bei NP-AL ; !+
> {+ibn:ibn NP-AL ; !+}
> {+la:la NP-AL ; !+}
> {+le:le NP-AL ; !+}
> {+les:les NP-AL ; !+}
> {+lès:lès NP-AL ; !+}
> ```

Oh, nevermind, these are forms you deleted, right?

**Comment by jonorthwash** [comment]
Why did you move to `U02BC`?  Uzbek Wikipedia uses `U02BB`, which I thought was the standard?  (I agree that the former looks better and is more appropriate, but it appears not to be what's used.)

**Comment by jonorthwash** [comment]
The roman numerals should probably have their own lexicon (pointed to directly from Root) and not be mixed into the main lexicon.

**Comment by jonorthwash** [comment]
Forms like this should be able to be generated automatically:
```
Chernishov:Chernishov  NP-COG-MF ; ! "" ! El++
Chernishova:Chernishova  NP-COG-MF ; ! "" ! El++
```
Forms like this appear to be patronymics, which can also be generated for `<m>` and `<f>` forms automatically:
```
Cholakovich:Cholakovich  NP-COG-MF ; ! "" ! El++
```

**Review by jonorthwash — CHANGES_REQUESTED** [review]
See above comments

**Comment by elmurod1202** [comment]
> I'm curious about these forms; what are they?
> 
> ```
> {+ad:ad NP-AL ; !+}
> {+al:al NP-AL ; !+}
> {+am:am NP-AL ; !+}
> {+bei:bei NP-AL ; !+
> {+ibn:ibn NP-AL ; !+}
> {+la:la NP-AL ; !+}
> {+le:le NP-AL ; !+}
> {+les:les NP-AL ; !+}
> {+lès:lès NP-AL ; !+}
> ```

These would be parts of NP-ORGS, no?. Why I added them is because, first of all, they exist in turkish.lexc(and all the time I consider tur.lexc as an ideal package and try to replicate what's there), second of all, they help improve the coverage. Was it wrong?(If so, I'll fix them from both Turkish and Uzbek)

**Comment by elmurod1202** [comment]
> Why did you move to `U02BC`? Uzbek Wikipedia uses `U02BB`, which I thought was the standard? (I agree that the former looks better and is more appropriate, but it appears not to be what's used.)

The shortest answer is: I didn't. Explanation: The apostrophe in <oʻ> and <gʻ> are U02BB and I converted all to that, there is another apostrophe in the alphabet: "tutuq belgisi"(phonetic glottal stop, <ъ> in cyrillic script) and itʻs U02BC,(Ex: aʼlo, maʼno, eʼzoz) I fixed them as well.

**Comment by elmurod1202** [comment]
> The roman numerals should probably have their own lexicon (pointed to directly from Root) and not be mixed into the main lexicon.

This way of announcing roman numerals was also copied from apertium-tur. Now I see what you(that was you, wasn't?) did in apertium-kaz/tat do deal with it (<(M | D | C | L | X | V | I)+> NUM-ROMAN ;). That looks more appropriate, I'll fix both tur and uzb if you confirm.

**Comment by jonorthwash** [comment]
> > I'm curious about these forms; what are they?
> > ```
> > {+ad:ad NP-AL ; !+}
> > {+al:al NP-AL ; !+}
> > {+am:am NP-AL ; !+}
> > {+bei:bei NP-AL ; !+
> > {+ibn:ibn NP-AL ; !+}
> > {+la:la NP-AL ; !+}
> > {+le:le NP-AL ; !+}
> > {+les:les NP-AL ; !+}
> > {+lès:lès NP-AL ; !+}
> > ```
> 
> These would be parts of NP-ORGS, no?. Why I added them is because, first of all, they exist in turkish.lexc(and all the time I consider tur.lexc as an ideal package and try to replicate what's there), second of all, they help improve the coverage. Was it wrong?(If so, I'll fix them from both Turkish and Uzbek)

1. I'm not sure it's a good idea to always copy other Apertium pairs.  `tur.lexc` is generally in better shape than `uzb.lexc`, but it's far from perfect.

2. I'm not sure if we need these forms, but it might not hurt.  I don't know.  Let's not worry about it now, maybe...

**Comment by jonorthwash** [comment]
> > Why did you move to `U02BC`? Uzbek Wikipedia uses `U02BB`, which I thought was the standard? (I agree that the former looks better and is more appropriate, but it appears not to be what's used.)
> 
> The shortest answer is: I didn't. Explanation: The apostrophe in <oʻ> and <gʻ> are U02BB and I converted all to that, there is another apostrophe in the alphabet: "tutuq belgisi"(phonetic glottal stop, <ъ> in cyrillic script) and itʻs U02BC,(Ex: aʼlo, maʼno, eʼzoz) I fixed them as well.

Oh!  I didn't know these two were meant to be encoded differently!  So, how do we know what the right encoding of them is?

**Comment by jonorthwash** [comment]
> > The roman numerals should probably have their own lexicon (pointed to directly from Root) and not be mixed into the main lexicon.
> 
> This way of announcing roman numerals was also copied from apertium-tur. Now I see what you(that was you, wasn't?) did in apertium-kaz/tat do deal with it (<(M | D | C | L | X | V | I)+> NUM-ROMAN ;). That looks more appropriate, I'll fix both tur and uzb if you confirm.

That was probably @IlnarSelimcan, and I assume that's right and should be okay to copy.

**Comment by elmurod1202** [comment]
> > > Why did you move to `U02BC`? Uzbek Wikipedia uses `U02BB`, which I thought was the standard? (I agree that the former looks better and is more appropriate, but it appears not to be what's used.)
> > 
> > 
> > The shortest answer is: I didn't. Explanation: The apostrophe in <oʻ> and <gʻ> are U02BB and I converted all to that, there is another apostrophe in the alphabet: "tutuq belgisi"(phonetic glottal stop, <ъ> in cyrillic script) and itʻs U02BC,(Ex: aʼlo, maʼno, eʼzoz) I fixed them as well.
> 
> Oh! I didn't know these two were meant to be encoded differently! So, how do we know what the right encoding of them is?

Wiki page on "Uzbek alphabet" says that U02BB is an apostrophe used for <oʻ> and <gʻ>, while U02BC is used for "tutuq belgisi"(aka glottal stop?).  I fixed all apostrophes in uzb.lexc in my branch to what they are supposed to be. Yet there is a problem still standing is that due to the lack of proper keyboard layout for Uzbek alphabet, those apostrophes appear in varieties of forms, it has to be solved though (Issue #2).

**Comment by ftyers** [comment]
> > > The roman numerals should probably have their own lexicon (pointed to directly from Root) and not be mixed into the main lexicon.
> > 
> > 
> > This way of announcing roman numerals was also copied from apertium-tur. Now I see what you(that was you, wasn't?) did in apertium-kaz/tat do deal with it (<(M | D | C | L | X | V | I)+> NUM-ROMAN ;). That looks more appropriate, I'll fix both tur and uzb if you confirm.
> 
> That was probably @IlnarSelimcan, and I assume that's right and should be okay to copy.

This might be problematic for the "entry beginning with whitespace" bug. You might want to use `[ ... ]` instead of `( ... )`.

**Review by jonorthwash — APPROVED** [review]

**Pull request merged**